make test - lookup test fails

running `make test` failes:

```
msperl@mslin confd]$ make test
Running tests...
ok      github.com/kelseyhightower/confd        0.009s
?       github.com/kelseyhightower/confd/backends       [no test files]
?       github.com/kelseyhightower/confd/backends/consul        [no test files]
?       github.com/kelseyhightower/confd/backends/dynamodb      [no test files]
?       github.com/kelseyhightower/confd/backends/env   [no test files]
?       github.com/kelseyhightower/confd/backends/etcd  [no test files]
?       github.com/kelseyhightower/confd/backends/etcdv3        [no test files]
?       github.com/kelseyhightower/confd/backends/file  [no test files]
?       github.com/kelseyhightower/confd/backends/rancher       [no test files]
?       github.com/kelseyhightower/confd/backends/redis [no test files]
?       github.com/kelseyhightower/confd/backends/ssm   [no test files]
?       github.com/kelseyhightower/confd/backends/vault [no test files]
?       github.com/kelseyhightower/confd/backends/zookeeper     [no test files]
?       github.com/kelseyhightower/confd/integration/ssm        [no test files]
?       github.com/kelseyhightower/confd/integration/zookeeper  [no test files]
?       github.com/kelseyhightower/confd/log    [no test files]
--- FAIL: TestTemplates (0.05s)
        template_test.go:734: ip lookup test: invalid StageFile. Expected 

                ip: 127.0.0.1

                , actual 

                ip: 127.0.0.1

                ip: ::1

FAIL
FAIL    github.com/kelseyhightower/confd/resource/template      0.056s
make: *** [test] Error 1

```

/etc/host looks like this:
```
[msperl@mslin confd]$ cat /etc/hosts
127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4 mslin
::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
```
I guess it is related to ipv6 support, as commenting out the ::1 line in /etc/hosts resolves that issue...

--- a/resource/template/template_funcs.go
+++ b/resource/template/template_funcs.go
@@ -184,7 +184,6 @@ func LookupIP(data string) []string {
 }
 
 func LookupIPV6(data string) []string {
-
 	addresses := make([]string, 0)
 	for _, ip := range LookupIP(data) {
 		if strings.Contains(ip, ":") {
@@ -195,7 +194,6 @@ func LookupIPV6(data string) []string {
 }
 
 func LookupIPV4(data string) []string {
-
 	addresses := make([]string, 0)
 	for _, ip := range LookupIP(data) {
 		if strings.Contains(ip, ".") {

--- a/resource/template/template_funcs.go
+++ b/resource/template/template_funcs.go
@@ -184,7 +184,7 @@ func LookupIP(data string) []string {
 }
 
 func LookupIPV6(data string) []string {
-	addresses := make([]string, 0)
+	var addresses []string
 	for _, ip := range LookupIP(data) {
 		if strings.Contains(ip, ":") {
 			addresses = append(addresses, ip)
@@ -194,7 +194,7 @@ func LookupIPV6(data string) []string {
 }
 
 func LookupIPV4(data string) []string {
-	addresses := make([]string, 0)
+	var addresses []string
 	for _, ip := range LookupIP(data) {
 		if strings.Contains(ip, ".") {
 			addresses = append(addresses, ip)

--- a/resource/template/template_funcs.go
+++ b/resource/template/template_funcs.go
@@ -33,6 +33,8 @@ func newFuncMap() map[string]interface{} {
 	m["replace"] = strings.Replace
 	m["trimSuffix"] = strings.TrimSuffix
 	m["lookupIP"] = LookupIP
+	m["lookupIPV4"] = LookupIPV4
+	m["lookupIPV6"] = LookupIPV6
 	m["lookupSRV"] = LookupSRV
 	m["fileExists"] = isFileExist
 	m["base64Encode"] = Base64Encode
@@ -179,6 +181,28 @@ func LookupIP(data string) []string {
 	}
 	sort.Strings(ipStrings)
 	return ipStrings
+}
+
+func LookupIPV6(data string) []string {
+
+	addresses := make([]string, 0)
+	for _, ip := range LookupIP(data) {
+		if strings.Contains(ip, ":") {
+			addresses = append(addresses, ip)
+		}
+	}
+	return addresses
+}
+
+func LookupIPV4(data string) []string {
+
+	addresses := make([]string, 0)
+	for _, ip := range LookupIP(data) {
+		if strings.Contains(ip, ".") {
+			addresses = append(addresses, ip)
+		}
+	}
+	return addresses
 }
 
 type sortSRV []*net.SRV

--- a/resource/template/template_test.go
+++ b/resource/template/template_test.go
@@ -109,7 +109,7 @@ type templateTest struct {
 	desc        string                  // description of the test (for helpful errors)
 	toml        string                  // toml file contents
 	tmpl        string                  // template file contents
-	expected    string                  // expected generated file contents
+	expected    interface{}                  // expected generated file contents
 	updateStore func(*TemplateResource) // function for setting values in store
 }
 
@@ -616,6 +616,58 @@ dir: /test/data
 		},
 	},
 	templateTest{
+		desc: "ipv4 lookup test",
+		toml: `
+[template]
+src = "test.conf.tmpl"
+dest = "./tmp/test.conf"
+keys = [
+    "/test/data",
+    "/test/data/abc",
+]
+`,
+		tmpl: `
+{{range lookupIPV4 "localhost"}}
+ip: {{.}}
+{{end}}
+`,
+		expected: `
+
+ip: 127.0.0.1
+
+`,
+		updateStore: func(tr *TemplateResource) {
+			tr.store.Set("/test/data", "parent")
+			tr.store.Set("/test/data/def", "child")
+		},
+	},
+	templateTest{
+		desc: "ipv6 lookup test",
+		toml: `
+[template]
+src = "test.conf.tmpl"
+dest = "./tmp/test.conf"
+keys = [
+    "/test/data",
+    "/test/data/abc",
+]
+`,
+		tmpl: `
+{{range lookupIPV6 "localhost"}}
+ip: {{.}}
+{{end}}
+`,
+		expected: `
+
+ip: ::1
+
+`,
+		updateStore: func(tr *TemplateResource) {
+			tr.store.Set("/test/data", "parent")
+			tr.store.Set("/test/data/def", "child")
+		},
+	},
+	templateTest{
 		desc: "ip lookup test",
 		toml: `
 [template]
@@ -631,11 +683,25 @@ keys = [
 ip: {{.}}
 {{end}}
 `,
-		expected: `
+		expected: [...]string{
+`
 
 ip: 127.0.0.1
 
 `,
+`
+
+ip: 127.0.0.1
+
+ip: ::1
+
+`,
+`
+
+ip: ::1
+
+`,
+},
 		updateStore: func(tr *TemplateResource) {
 			tr.store.Set("/test/data", "parent")
 			tr.store.Set("/test/data/def", "child")
@@ -749,8 +815,18 @@ func ExecuteTestTemplate(tt templateTest, t *testing.T) {
 	if err != nil {
 		t.Errorf(tt.desc + ": failed to read StageFile: " + err.Error())
 	}
-	if string(actual) != tt.expected {
-		t.Errorf(fmt.Sprintf("%v: invalid StageFile. Expected %v, actual %v", tt.desc, tt.expected, string(actual)))
+	switch tt.expected.(type) {
+	case string:
+		if string(actual) != tt.expected.(string) {
+			t.Errorf(fmt.Sprintf("%v: invalid StageFile. Expected %v, actual %v", tt.desc, tt.expected, string(actual)))
+		}
+	case []string:
+		for _, expected := range tt.expected.([]string) {
+			if string(actual) == expected {
+				break
+			}
+		}
+		t.Errorf(fmt.Sprintf("%v: invalid StageFile. Possible expected values %v, actual %v", tt.desc, tt.expected, string(actual)))
 	}
 }
 

--- a/resource/template/template_test.go
+++ b/resource/template/template_test.go
@@ -109,7 +109,7 @@ type templateTest struct {
 	desc        string                  // description of the test (for helpful errors)
 	toml        string                  // toml file contents
 	tmpl        string                  // template file contents
-	expected    interface{}                  // expected generated file contents
+	expected    interface{}             // expected generated file contents
 	updateStore func(*TemplateResource) // function for setting values in store
 }
 
@@ -684,24 +684,24 @@ ip: {{.}}
 {{end}}
 `,
 		expected: [...]string{
-`
+			`
 
 ip: 127.0.0.1
 
 `,
-`
+			`
 
 ip: 127.0.0.1
 
 ip: ::1
 
 `,
-`
+			`
 
 ip: ::1
 
 `,
-},
+		},
 		updateStore: func(tr *TemplateResource) {
 			tr.store.Set("/test/data", "parent")
 			tr.store.Set("/test/data/def", "child")

--- a/resource/template/template_test.go
+++ b/resource/template/template_test.go
@@ -657,11 +657,15 @@ keys = [
 ip: {{.}}
 {{end}}
 `,
-		expected: `
-
+		expected: [...]string{
+			`
 ip: ::1
 
 `,
+			`
+
+`,
+},
 		updateStore: func(tr *TemplateResource) {
 			tr.store.Set("/test/data", "parent")
 			tr.store.Set("/test/data/def", "child")

